logstash-core/lib/logstash/compiler.rb

```diff
@@ -7,7 +7,9 @@ module LogStash; class Compiler
   include ::LogStash::Util::Loggable
 
   def self.compile_sources(sources_with_metadata, support_escapes)
-    graph_sections = sources_with_metadata.map do |swm|
+     graph_sections = sources_with_metadata.select do |swm|
+       swm.text.match(/\A\s*\Z/).nil?
```

Comment:
are we just looking for non-whitespaced strings? maybe using `reject` instead of `select`, with `swm.text.empty?` makes it easier to understand:

```ruby
     graph_sections = sources_with_metadata.reject do |swm|
        swm.text.empty?
     end.map do |swm|
        # ..
     end
```

Alternatively, do you think `compile_graph` should be able to compile `SourceWithMetadata` objects with empty sources?

Reply:
`empty?` is not enough because we have to consider also spaces and newlines as empty strings `"  \r\n\r\n".empty? != true` so that is the motivation for this regex. I think to make it more readable I should move in a separate method.
To make `compile_graph` able to compile also empty string we have 2 choices:
- put the empty check into `compile_imperative` and the open question is what king of AST we have to return?
- change LSCL grammar to return an AST with "EmptySynthaxNode" and manage it
I think that both make the change more harder. I would prefer the filtering way, probably separating the logic into own method for readability

Reply:
I've extracted the method and used the `reject` instead of `select`, let me know if it is better